polkadot-protocol/basics/randomness.md

```diff
@@ -1,4 +1,76 @@
+
+These numbers rely on an initial "seed", which can come from external sources like [atmospheric noise](https://www.random.org/randomness/){target=\_blank}, [heart rates](https://mdpi.altmetric.com/details/47574324){target=\_blank}, or even [lava lamps](https://en.wikipedia.org/wiki/Lavarand){target=\_blank}. While this may seem random, given the same "seed", the same sequence of numbers will always be generated.
+
+In a global blockchain network, relying on real-world entropy for randomness isn’t feasible because these inputs vary by time and location. If nodes use different inputs, blockchains can fork. Hence, real-world randomness isn't suitable for use as a seed in blockchain systems.
+
+Currently, two primary methods for generating randomness in blockchains are used: `RANDAO` and `VRF` (Verifiable Random Function). Polkadot adopts the `VRF` approach for its randomness.
+
+## VRF
+
+A **Verifiable Random Function (VRF)** is a cryptographic function that generates a random number along with a proof that ensures the number was produced by the submitter. This proof allows anyone to verify the validity of the random number.
+
+Polkadot's VRF is similar to the one used in [**Ouroboros Praos**](https://eprint.iacr.org/2017/573.pdf){target=\_blank}, which secures randomness for block production in systems like [BABE](link to BABE page) (Polkadot’s block production mechanism). 
+
+The key difference is that Polkadot's VRF doesn’t rely on a central clock—avoiding the issue of whose clock to trust. Instead, it uses its own past results and slot numbers to simulate time and determine future outcomes.
+
+**Here's how VRF works:**
+
+Slots on Polkadot are discrete units of time, each lasting six seconds, and can potentially hold a block. Multiple slots form an epoch, with 2400 slots making up one four-hour epoch.
+
+In each slot, validators execute a **"die roll"** using a **Verifiable Random Function (VRF)**. The VRF uses three inputs:
+
+1. A "secret key", unique to each validator, used for the die roll.
+2. An epoch randomness value, derived from the hash of VRF outputs from blocks two epochs ago (N-2), so past randomness influences the current epoch (N).
+3. The current slot number.
+
+This process helps maintain fair randomness across the network.
+
+Here is a graphical representation
+
+![](/images/polkadot-protocol/basics/blocks-transactions-fees/randomness/slots-epochs.webp)
+
+The `VRF` produces two outputs: a `RESULT` (the random number) and a `PROOF` (verifying that the number was generated correctly).
+
+The `RESULT` is checked by validator against a threshold set by the protocol. If it's below the threshold, the validator becomes a candidate for block production in that slot. 
+
+The validator then attempts to create a block, submitting it along with the `PROOF` and `RESULT`.
+
```

Comment:
@kapetan3sid I'm unsure if this overlaps too much with the content you wrote, but IMHO, this seems like something cool to have for those who are getting onboarded with this topic. Feel free to modify it or discard it in case you disagree

```suggestion
So, VRF can be expressed like:

`(RESULT, PROOF) = VRF(SECRET, EPOCH_RANDOMNESS_VALUE, CURRENT_SLOT_NUMBER)`

Put simply, performing a "VRF roll" generates a random number along with proof that the number was genuinely produced and not arbitrarily chosen.


After executing the VRF, the RESULT is compared to a protocol-defined THRESHOLD. If the RESULT is below the THRESHOLD, the validator becomes a valid candidate to propose a block for that slot. Otherwise, the validator skips the slot.

As a result, there may be multiple validators eligible to propose a block for a slot. In this case, the block accepted by other nodes will prevail, provided it is on the chain with the latest finalized block as determined by the GRANDPA finality gadget. It's also possible for no block producers to be available for a slot, in which case the AURA consensus takes over. AURA is a fallback mechanism that randomly selects a validator to produce a block, running in parallel with BABE and only stepping in when no block producers exist for a slot. Otherwise, it remains inactive.

```